client/src/components/TravelBar/TravelBar.jsx

```diff
@@ -1,61 +1,77 @@
-import React, {useEffect, useContext, useState} from 'react';
-import moment from 'moment';
-import {Context} from '../../state-management/Store';
-import TripInfoObject from './TripInfoObject.js';
-import './TravelBar.css';
-import SaveTrip from './SaveTrip';
-import {DatePicker} from 'antd';
-import SearchLocationInput from './SearchLocationInput.js';
-import TravelBarOptions from './TravelBarOptions.jsx';
+import React, { useEffect, useContext, useState } from "react";
+import moment from "moment";
+import { Context } from "../../state-management/Store";
+import TripInfoObject from "./TripInfoObject.js";
+import "./TravelBar.css";
+import SaveTrip from "./SaveTrip";
+import { DatePicker } from "antd";
+import SearchLocationInput from "./SearchLocationInput.js";
+import TravelBarOptions from "./TravelBarOptions.jsx";
 
 //should select buisness
 
 const TravelBar = () => {
-
   var [state, dispatch] = useContext(Context);
-  var [tripInfo, setTripInfo] = useState({...state.tripInfo})
+  var [tripInfo, setTripInfo] = useState({ ...state.tripInfo });
 
-    const disabledStartDate = (current) => {
+  const disabledStartDate = (current) => {
     let endDate = state.tripInfo.endDate;
     if (endDate || current) {
-       return (current && current > moment(endDate, "YYYY-MM-DD")) || (current && current < moment().endOf('day'))
+      return (
+        (current && current > moment(endDate, "YYYY-MM-DD")) ||
+        (current && current < moment().endOf("day"))
+      );
     }
-  }
+  };
   const disabledEndDate = (current) => {
     let startDate = state.tripInfo.startDate;
     if (startDate || current) {
-       return (current && current < moment(startDate, "YYYY-MM-DD")) || (current && current < moment().endOf('day'))
+      return (
+        (current && current < moment(startDate, "YYYY-MM-DD")) ||
+        (current && current < moment().endOf("day"))
+      );
     }
-  }
+  };
 
-  const handleTripInfoChange = (date, dateString, name ) => {
-    var tripInfoClone = {...tripInfo};
+  const handleTripInfoChange = (date, dateString, name) => {
+    var tripInfoClone = state.tripInfo;
```

Comment:
will this will assign tripInfoClone to be state.tripInfo? not copy it, but actually be the same object?

Reply:
Yes. You are updating the global state with the location when it is input but, the local state retains the default location of Austin. So at this point above if you copy the local state it resets the global state back to Austin as we saw yesterday. Doing it this way copies the global state to the local state when the startDate is updated. @Gledinburgh Let me know if you'd like to see this in action. Happy to take a look at it together.